operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -105,15 +111,45 @@ type RedpandaReconciler struct {
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *RedpandaReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager) error {
+	enqueueClusterFromNodePool := handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, o client.Object) []reconcile.Request {
+		pool := o.(*redpandav1alpha2.NodePool)
+		return []reconcile.Request{{
+			NamespacedName: types.NamespacedName{
+				Name:      pool.Spec.ClusterRef.Name,
+				Namespace: pool.Namespace,
+			},
+		}}
+	})
+
 	builder := ctrl.NewControllerManagedBy(mgr)
 
 	if err := r.LifecycleClient.WatchResources(builder, &redpandav1alpha2.Redpanda{}); err != nil {
 		return err
 	}
 
+	if r.UseNodePools {
+		builder.Watches(&redpandav1alpha2.NodePool{}, enqueueClusterFromNodePool)
+	}
```

Comment:
Just to have one less flag to juggle, what would you think about checking the discovery API to see if the nodepools CRD is installed? It seems like this is the only place `UseNodePools` is referenced?

Reply:
Yeah, and when fetching the pools to attach to the cluster. In general I’m fine keeping this flagged as the flag is meant to be temporary while the feature is “experimental”